app/components/Navigation.tsx

```diff
@@ -0,0 +1,101 @@
+import { memo, useState, Fragment, useMemo } from "react";
+import Home from "~/components/Home";
+import Profile from "~/components/Profile";
+import { useAuth0 } from "@auth0/auth0-react";
+import { Menu, Transition } from '@headlessui/react'
+
+//TODO: fix style of profile picture (maybe expand the image when the menu is opened as well)
+
+const Navigation = memo(function Nav({ apiBaseURL }: { apiBaseURL: string }){
+    const [darkMode, setDarkMode] = useState(true);
+    const [pageChanged, setPageChanged] = useState(true);
+
+    const changePage = (item: number) => {
+        switch(item){
+            case 1:
+                setPageChanged(true);
+                break;
+            case 2:
+                setPageChanged(false);
+                break;
+            case 3:
+                setDarkMode(darkMode => !darkMode);
+                break;  
+            case 4:
+                isAuthenticated ? logout() : loginWithRedirect();
+                break;       
+        }
+    }
+
+    const { loginWithRedirect, logout, user, isAuthenticated, isLoading } = useAuth0();
+
+    const userProfile = {
+        name: user?.name ?? "John Doe",
+        email: user?.email ?? 'johndoe@example.com',
+        imageUrl: isAuthenticated ? user?.picture : 'https://www.vecteezy.com/vector-art/550980-user-icon-vector',
+    }
+
+    const userNavigation = [
+        { name: 'Home', page: 1 },
+        { name: 'Profile', page: 2 },
+        { name: darkMode ? 'Light Mode' : 'Dark Mode', page: 3},
+        { name: isAuthenticated ? 'Sign in' : 'Sign out', page: 4},
+    ]
+      
+    function classNames(...classes: any[]) {
+        return classes.filter(Boolean).join(' ')
+    }
+
+    return (
+        <>  
+        <div className={darkMode ? "dark dark:bg-slate-800" : "bg-slate-100"}>
```

Comment:
I think if `color-scheme` is set at a higher level to control the toggle, this could have a single set of classes for each lighting scheme.